internal/zero/cmd/import_ui.go

```diff
@@ -0,0 +1,553 @@
+		"certificate_authority": {Note: noteCertificate(1)},
+		"certificates":          {Note: noteCertificate(len(cfg.GetSettings().GetCertificates()))},
+		"downstream_mtls.crl":   {Unsupported: true, Note: noteFeatureNotYetAvailable},
+		"downstream_mtls.ca":    {Note: noteCertificate(1)},
+	}
+	if dm := cfg.GetSettings().GetDownstreamMtls(); dm != nil {
+		if dm.Enforcement != nil {
+			switch *dm.Enforcement {
+			case configpb.MtlsEnforcementMode_POLICY:
+			case configpb.MtlsEnforcementMode_POLICY_WITH_DEFAULT_DENY:
+			case configpb.MtlsEnforcementMode_REJECT_CONNECTION:
+				// this is a special case - zero does not support this mode, but we cannot continue
+				// with a partial import because it fundamentally changes the behavior of all routes
+				// and policies in the system
+				log.Fatal().Msg("downstream mtls enforcement mode 'reject_connection' is not supported")
+			}
+		}
+	}
+	if cfg.GetSettings().GetServices() != "all" {
+		m["services"] = ImportHints{Ignored: true, Note: `only "all" is supported`}
+	}
+	if cfg.GetSettings().GetCodecType() != http_connection_managerv3.HttpConnectionManager_AUTO {
+		m["codec_type"] = ImportHints{Ignored: true, Note: `only "auto" is supported`}
+	}
+	return m
+}
+
+type ImportUI struct {
+	form             *huh.Form
+	selectedSettings []string
+	selectedRoutes   []string
+}
+
+func NewImportUI(cfg *configpb.Config, quotas *cluster_api.ConfigQuotas) *ImportUI {
+	settingsImportHints := computeSettingsImportHints(cfg)
+
+	presentSettings := fieldmasks.Leaves(
+		fieldmasks.Diff(
+			config.NewDefaultOptions().ToProto().GetSettings().ProtoReflect(),
+			cfg.GetSettings().ProtoReflect(),
+		),
+		cfg.Settings.ProtoReflect().Descriptor(),
+	)
+	slices.Sort(presentSettings.Paths)
+	settingsOptions := huh.NewOptions(presentSettings.Paths...)
+
+	ui := &ImportUI{
+		selectedSettings: slices.Clone(presentSettings.Paths),
+	}
+
+	for i, value := range presentSettings.Paths {
+		if hints, ok := settingsImportHints[value]; ok {
+			switch {
+			case hints.Ignored:
+				note := ""
+				if hints.Note != "" {
+					note = fmt.Sprintf(": %s", hints.Note)
+				}
+				settingsOptions[i].Key = fmt.Sprintf("\x1b[9m%s\x1b[29m \x1b[2m(ignored%s)\x1b[22m", settingsOptions[i].Key, note)
+				ui.selectedSettings[i] = ""
+			case hints.Unsupported:
+				note := ""
+				if hints.Note != "" {
+					note = fmt.Sprintf(": %s", hints.Note)
+				}
+				settingsOptions[i].Key = fmt.Sprintf("\x1b[9m%s\x1b[29m \x1b[2m(unsupported%s)\x1b[22m", settingsOptions[i].Key, note)
+				ui.selectedSettings[i] = ""
+			case hints.Secret:
+				settingsOptions[i].Key += " \x1b[2m(secret)\x1b[22m"
+			default:
+				if hints.Note != "" {
+					settingsOptions[i].Key += fmt.Sprintf(" \x1b[2m(%s)\x1b[22m", hints.Note)
+				}
+			}
+		}
+	}
+	ui.selectedSettings = slices.DeleteFunc(ui.selectedSettings, func(s string) bool {
+		return s == ""
+	})
+	settingsSelect := huh.NewMultiSelect[string]().
```

Comment:
I can see why we might want a routes chooser (in case you have more routes than the quotas allow for), but I'm not sure I understand why you would want to import some settings but not others.

Should we just import all settings that are supported in Zero?

Reply:
We want to give the user a way to opt-out of uploading some settings that can contain sensitive data, like secrets or idp credentials. There are many unsupported or ignored settings as well, and this is a convenient way to display those.

<details>
<summary>screenshot</summary>
<img src="https://github.com/user-attachments/assets/ec59099e-90de-441d-8aeb-d5272d58b313"/>
</details>